tests/phpunit/unit-tests/functions/class-llms-test-functions-page.php

```diff
@@ -41,6 +41,108 @@ public function test_llms_confirm_payment_url() {
 
 	}
 
+	/**
+	 * Test llms_get_endpoint_url() when pretty permalinks are disabled.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function test_llms_get_endpoint_url_no_pretty_permalinks() {
+
+		LLMS_Install::create_pages();
+
+		$permalink = get_permalink( llms_get_page_id( 'myaccount' ) );
+		$this->go_to( $permalink );
+
+		foreach ( llms()->query->get_query_vars() as $var => $slug ) {
+
+			$this->assertEquals( "{$permalink}&{$slug}", llms_get_endpoint_url( $var ) );
+			$this->assertEquals( "{$permalink}&{$slug}=test", llms_get_endpoint_url( $var, 'test' ) );
+			$this->assertEquals( "https://fake.tld/?{$slug}=1", llms_get_endpoint_url( $var, 1, 'https://fake.tld/' ) );
+
+		}
+
+	}
+
+	/**
+	 * Test llms_get_endpoint_url() when pretty permalinks are enabled with a trailing slash.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function test_llms_get_endpoint_url_with_trailing_slash_pretty_permalink() {
+
+		$orig_permastruct = get_option( 'permalink_structure' );
+
+		LLMS_Install::create_pages();
+
+		update_option( 'permalink_structure', '/%postname%/' );
+		global $wp_rewrite;
```

Comment:
Traditionally in PHP, use of the global keyword happens at the start of the function.